executor/adapter.go

```diff
@@ -155,17 +164,21 @@ type ExecStmt struct {
 	isPreparedStmt bool
 }
 
-// OriginText implements ast.Statement interface.
+// OriginText returns original statement as a string.
 func (a *ExecStmt) OriginText() string {
 	return a.Text
 }
 
-// IsPrepared implements ast.Statement interface.
+// IsPrepared return true if stmt is already prepared.
 func (a *ExecStmt) IsPrepared() bool {
 	return a.isPreparedStmt
 }
 
-// IsReadOnly implements ast.Statement interface.
+// IsReadOnly return true if stmt is read only.
+// First fetch a to be checked statement from ExecStme
+// and update it if plan is a execute plan.
```

Comment:
This detailed info should be commented in the function, not as the comment of the function.

Reply:
In my opinion, a function's comment should describe basic logic of the function. 
here is a example https://github.com/golang/go/blob/master/src/fmt/print.go#L411.